packages/hop-node/src/watchers/classes/Bridge.ts

```diff
@@ -527,6 +534,21 @@ export default class Bridge extends ContractBase {
     return tx
   }
 
+  @rateLimitRetry
+  async getStateUpdateStatus (stateUpdateAddress: string, chainId: number): Promise<number> {
+    const abi = ['function currentState(address,uint256)']
+    const ethersInterface = new ethersUtils.Interface(abi)
+    const data = ethersInterface.encodeFunctionData(
+      'currentState', [this.l1CanonicalTokenAddress, chainId]
+    )
+    const tx = {
+      to: stateUpdateAddress,
+      data
+    }
+    const res: string = await this.contract.provider.call(tx)
+    return Number(res)
```

Comment:
maybe a dumb question but would `res` here ever be a big number value that `Number` won't be a be to handle? (ie does it need to be wrapped in a big number type and then converted to Number)

Reply:
Technically yes if we pass in a number that is outside of our options for state code. In practice, the result should only every be 1, 2, 3, or 4 (or 5, 6, etc. if we add more states). The `res` for 1, for example, looks like `0x0000000000000000000000000000000000000001`.

If `0xfff...` is the currents state, then I do think this would break. That should never happen but I'd be happy to add a check here (or on the contract) that would confirm this. LMK what you think.

Reply:
The way it currently is works as expected, so we can leave as is

```
Number('0x0000000000000000000000000000000000000001') // 1
Number('0x0000000000000000000000000000000000000002') // 2
```